File: src/test/java/com/automation/um/pages/DashboardPage.java
package com.automation.um.pages;

import com.automation.um.config.Config;
import org.openqa.selenium.WebDriver;

public class DashboardPage extends BasePage {
  public DashboardPage(WebDriver driver) {
    super(driver);
  }

  public void waitForLoaded() {
    el("stat-total-users");
    wait.until(d -> {
      String text = d.findElement(testId("stat-total-users")).getText().trim();
      return text.matches("\\d+");
    });
  }

  public int getTotalUsersCount() {
    return Integer.parseInt(el("stat-total-users").getText().trim());
  }

  public boolean isOnDashboard() {
    return driver.getCurrentUrl().replaceAll("/$", "").endsWith(Config.BASE_URL.replaceAll("/$", ""))
        || driver.getCurrentUrl().endsWith("/")
        && isVisible("stat-total-users");
  }
}

File: playwright/pages/DashboardPage.js
import { BasePage } from "./BasePage.js";
import { BASE_URL } from "../config/config.js";

export class DashboardPage extends BasePage {
  async waitForLoaded() {
    await this.el("stat-total-users");
    await this.page.waitForFunction(() => {
      const el = document.querySelector("[data-testid='stat-total-users']");
      if (!el) return false;
      return /^\d+$/.test(el.textContent?.trim() || "");
    });
  }

  async getTotalUsersCount() {
    const text = await this.testId("stat-total-users").textContent();
    return Number.parseInt(text?.trim() || "0", 10);
  }

  async isOnDashboard() {
    const url = this.page.url().replace(/\/$/, "");
    const base = BASE_URL.replace(/\/$/, "");
    return (url === base || url.endsWith("/")) && (await this.isVisible("stat-total-users"));
  }
}
